### DISCUSSION TITLE
Searching through files

### DISCUSSION THREAD

**Discussion by whittede** [Natizyskunk/vscode-sftp#280] [General]
In the SFTP extension browser, I can view all files that are on the remote server, as well as their contents, without downloading them.

I can use Ctrl+F on my keyboad to search through the _names_ of the files on the remote server.
I can use Ctrl+Shift+F on my keyboard to search through the _contents_ of all files that are locally on my computer.

**Is there a way to search through the contents of all files that are on the remote server without downloading all of those files to my local computer?** Something that has the same functionality as the Ctrl+Shift+F command, but for the SFTP file explorer.

**Comment by Natizyskunk** [comment]
@whittede, So basically a global search functionality for the remote explorer ?

**Comment by satiromarra** [comment]
@whittede @Natizyskunk 
I'm sorry to say that you can't search for files for content through ftp/sftp/ftps.
However, if the configuration is sftp you can search remotely using ssh:
`ssh user@host -i ~/.ssh/id_rsa "find $folder -type f -exec grep \"$keywords\" -ls {} +"`
This will return a list of files or nothing.
Cons: It may require a password or some type of validation at the beginning of the connection.
